src/plugins/workspace/server/routes/index.ts

```diff
@@ -218,4 +218,59 @@ export function registerRoutes({
       return res.ok({ body: result });
     })
   );
+  router.get(
+    {
+      path: `${WORKSPACES_API_BASE_URL}/_allowed_types`,
+      validate: false,
+    },
+    async (context, req, res) => {
+      const allowedTypes = context.core.savedObjects.typeRegistry
+        .getImportableAndExportableTypes()
+        .map((type) => type.name);
+
+      return res.ok({
+        body: {
+          types: allowedTypes,
+        },
+      });
+    }
+  );
+  router.post(
+    {
+      path: `${WORKSPACES_API_BASE_URL}/_move_objects`,
+      validate: {
+        body: schema.object({
+          objects: schema.maybe(
+            schema.arrayOf(
+              schema.object({
+                type: schema.string(),
+                id: schema.string(),
+                workspaces: schema.arrayOf(schema.string()),
+                attributes: schema.recordOf(schema.string(), schema.any()),
+              })
+            )
+          ),
+          sourceWorkspaceId: schema.string(),
+          targetWorkspaceId: schema.string(),
+        }),
+      },
+    },
+    router.handleLegacyErrors(async (context, req, res) => {
+      const savedObjectsClient = context.core.savedObjects.client;
+      const { objects, sourceWorkspaceId, targetWorkspaceId } = req.body;
+      const objectsToMove = objects.map((obj) => {
+        const sourceIndex = obj.workspaces.indexOf(sourceWorkspaceId);
+        if (sourceIndex !== -1) {
+          obj.workspaces.splice(sourceIndex, 1);
+        }
+        const targetIndex = obj.workspaces.indexOf(targetWorkspaceId);
+        if (targetIndex === -1) {
```

Comment:
Shall we skip move related saved objects if it's already in target workspace?

Reply:
yes, I will add a logic of filtering target workspace ID in api.